test/jdk/javax/swing/text/html/HTMLDocument/bug4209280.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 1999, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @bug 4209280
+ * @key headful
+ * @summary Tests that no exception is thrown on unknown HTML tag
+ */
+
+public class bug4209280 {
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            JFrame jFrame = new JFrame("Unknown HTML tag Test");
+            String html = "<html><bold>Foo</bold></html>";
+            JLabel label = new JLabel(html);
```

Comment:
I wonder if the test could be headless. I presume the exception is thrown when HTML is parsed. Does it happen in constructor? Does it happen when the label is added to a container?

Is it enough to create the label?

Reply:
I believe the original issue occurred when invalid html tags were set in the constructor -`new JLabel(html)` [JDK-4245889](https://bugs.openjdk.org/browse/JDK-4245889).

For making it headless I think it is probably sufficient to just create the label with invalid html tag then?

Reply:
I think so.

You can also run the test in jdk1.2.0 or jdk1.2.2 to ensure it still reproduces the problem and then confirm jdk1.3.0 doesn't reproduce it. Without `invokeAndWait` and the lambda expression, it should compile fine.

Reply:
The original issue occurred at the constructor, confirmed by checking it using jdk1.2.2
Test has been made headless and reduced to following two lines. 

```
String html = "<html><bold>Foo</bold></html>";
new JLabel(html);
```